core/frontend/src/store/notifications.ts

```diff
@@ -0,0 +1,72 @@
+import { Module, Mutation, VuexModule } from 'vuex-module-decorators'
+
+import store from '@/store'
+import { CumulatedNotification, Notification, NotificationLevel } from '@/types/notifications'
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'notification_store',
+})
+
+@Module
+export default class NotificationStore extends VuexModule {
+  notifications: Notification[] = []
+
+  cumulated_notifications: CumulatedNotification[] = []
+
+  dismissed_notifications: Notification[] = []
+
+  @Mutation
+  pushNotification(notification: Notification): void {
+    this.notifications.push(notification)
+
+    switch (notification.level) {
+      case NotificationLevel.Success:
+        console.log(notification.message)
+        break
+      case NotificationLevel.Error:
+        console.error(notification.message)
+        break
+      case NotificationLevel.Info:
+        console.info(notification.message)
+        break
+      case NotificationLevel.Warning:
+        console.warn(notification.message)
+        break
+      case NotificationLevel.Critical:
+        console.exception(notification.message)
+        break
+      default:
+        break
```

Comment:
Can we add a message here when a invalid or not exhaustive switch case happen ?
E.g: A new notification level was added but we have not updated this part of the code

Reply:
Good one! Adding.